File: frontend/src/features/projects/services/clientService.ts
import apiClient from '../../../api/apiClient';
import type { PagedResponse } from '../../../shared/types/common.types';
import type { Client, CreateClientRequest, UpdateClientRequest } from '../types/project.types';

export const clientService = {
  getAll: async (page = 0, size = 20): Promise<PagedResponse<Client>> => {
    const response = await apiClient.get('/clients', { params: { page, size } });
    return response.data;
  },

  getAllActive: async (): Promise<Client[]> => {
    const response = await apiClient.get('/clients/active');
    return response.data;
  },

  getById: async (id: string): Promise<Client> => {
    const response = await apiClient.get(`/clients/${id}`);
    return response.data;
  },

  create: async (data: CreateClientRequest): Promise<Client> => {
    const response = await apiClient.post('/clients', data);
    return response.data;
  },

  update: async (id: string, data: UpdateClientRequest): Promise<Client> => {
    const response = await apiClient.put(`/clients/${id}`, data);
    return response.data;
  },

  delete: async (id: string): Promise<void> => {
    await apiClient.delete(`/clients/${id}`);
  },
};

File: backend/src/main/java/com/turntabl/bonarda/domain/project/service/ClientService.java
package com.turntabl.bonarda.domain.project.service;

import com.turntabl.bonarda.domain.project.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface ClientService {
    ClientDto create(CreateClientRequest request);
    ClientDto update(UUID publicId, UpdateClientRequest request);
    ClientDto getById(UUID publicId);
    Page<ClientDto> getAll(Pageable pageable);
    List<ClientDto> getAllActive();
    void delete(UUID publicId);
}
